src/mediapipe_internal/pythonnoderesource.cpp

```diff
@@ -37,6 +37,32 @@ PythonNodeResource::PythonNodeResource() {
     this->nodeResourceObject = nullptr;
 }
 
+Status PythonNodeResource::finalize() {
+    if (this->nodeResourceObject) {
+        py::gil_scoped_acquire acquire;
+        try {
+            if (!py::hasattr(*nodeResourceObject.get(), "finalize")) {
+                SPDLOG_DEBUG("Python node resource does not have a finalize method. Python node path {} ", this->pythonNodeFilePath);
+                return StatusCode::OK;
+            }
+
+            py::bool_ success = nodeResourceObject.get()->attr("finalize")();
```

Comment:
I'm not sure if we should expect any value from finalize(). Unloading happens anyway so OVMS will not do anything with such information except for logging it. Whether finalize will complete successfully, return true, false or None, the node will get destroyed and won't be functional anymore.